packages/cardano-services/src/Asset/CardanoTokenRegistry.ts

```diff
@@ -148,7 +148,18 @@ export class CardanoTokenRegistry implements TokenMetadataService {
         }
       }
     } catch (error) {
-      throw toProviderError(error, 'while fetching metadata from token registry');
+      // Propagete only the Provider errors
+      // We are interested only from network errors while fetching from external token metadata server
+      if (error instanceof ProviderError) {
+        throw error;
+      }
+
+      // Return `undefined` in case of HTTP error while fetching from Token Metadata server
```

Comment:
On second look at this, I think the handling should be done within `DbSyncAssetProvider`, as that's the more appropriate context to assign the value of `undefined`. Here I would just ensure the errors being rethrown are suitable for interpretation.

Reply:
Agree that the error handling could be lifted one level up in `DbSyncAssetProvider`. The necessary distinction of errors in our case is already suitable for interpretation since we have this [try/catch](https://github.com/input-output-hk/cardano-js-sdk/blob/039e0f0069d493cf226c0b37aeec34464a5d96ed/packages/cardano-services/src/Asset/CardanoTokenRegistry.ts#L131-L148)  for all **provider** related errors and interpret all others as **HTTP** related ones thrown by [axios client](https://github.com/input-output-hk/cardano-js-sdk/blob/039e0f0069d493cf226c0b37aeec34464a5d96ed/packages/cardano-services/src/Asset/CardanoTokenRegistry.ts#L125-L128)

Reply:
b856d33aa8738e7d6cd6c7685c7a9689cea2de6d